pandaserver/dataservice/event_picker.py

```diff
@@ -0,0 +1,276 @@
+            ignoreError : bool
+                Whether to ignore errors.
+        """
+        self.task_buffer = taskBuffer
+        self.site_mapper = siteMapper
+        self.ignore_error = ignoreError
+        self.event_picking_file_name = evpFileName
+        self.token = datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None).isoformat(" ")
+        # logger
+        self.logger = LogWrapper(_logger, self.token)
+        self.pd2p = dyn_data_distributer.DynDataDistributer([], self.site_mapper, token=" ")
+        self.user_dataset_name = ""
+        self.creation_time = ""
+        self.params = ""
+        self.locked_by = ""
+        self.event_picking_file = None
+        self.user_task_name = ""
+        self.user_dn = ""
+        # JEDI
+        self.jedi_task_id = None
+
+    # end with error
+    def end_with_error(self, message: str):
+        """
+        Ends the event picker with an error.
+
+        This method is called when an error occurs during the event picking process. It logs the error message,
+        unlocks and closes the event picking file, and removes it if the error is not to be ignored. It then uploads
+        the log and updates the task status in the task buffer.
+
+        Parameters:
+            message (str): The error message to be logged.
+        """
+        self.put_log(message, "error")
+        # unlock evp file
+        try:
+            fcntl.flock(self.event_picking_file.fileno(), fcntl.LOCK_UN)
+            self.event_picking_file.close()
+            if not self.ignore_error:
+                # remove evp file
+                os.remove(self.event_picking_file_name)
+        except Exception:
+            pass
+        # upload log
+        if self.jedi_task_id is not None:
+            out_log = self.upload_log()
+            self.task_buffer.updateTaskErrorDialogJEDI(self.jedi_task_id, "event picking failed. " + out_log)
+            # update task
+            if not self.ignore_error:
+                self.task_buffer.updateTaskModTimeJEDI(self.jedi_task_id, "tobroken")
+            self.put_log(out_log)
+        self.put_log(f"end {self.event_picking_file_name}")
+
+    # put log
+    def put_log(self, msg: str, msg_type: str = "debug"):
+        """
+        Logs a message with a specified type.
+
+        This method logs a message with a specified type. The type can be either "debug" or "error".
+        If the msg_type is "error", the message is logged as an error. Otherwise, it is logged as a debug message.
+
+        Parameters:
+            msg (str): The message to be logged.
+            msg_type (str): The type of the log. It can be either "debug" or "error". Default is "debug".
+
+        Returns:
+            None
+        """
+        tmp_msg = msg
+        if msg_type == "error":
+            self.logger.error(tmp_msg)
+        else:
+            self.logger.debug(tmp_msg)
+
+    # upload log
+    def upload_log(self) -> str:
+        """
+        Uploads the log.
```

Comment:
upload_log sends the file to panda_server (CC @tmaeno)
- To make the comment more useful, who consumes the file? 
- Why do a http call to panda server, instead of just writing the file to the cache_dir? Maybe there should be a Client.write_log function?